lightning/src/ln/channelmanager.rs

```diff
@@ -5278,6 +5299,7 @@ where
 										}, skimmed_fee_msat, ..
 									},
 								}) => {
+									let logger = WithChannelContext::from(&self.logger, &chan.context, Some(payment_hash));
```

Comment:
In this method in the `check_total_value` macro and elsewhere, could use the contextual logger rather than `self.logger`. Those logs do currently contain the payment hash anyway, but I think they may print bytes instead of hex.

Reply:
added https://github.com/lightningdevkit/rust-lightning/pull/2930/commits/232d6af3ffa860705b8edd8136d5fff1ba573c3a and https://github.com/lightningdevkit/rust-lightning/pull/2930/commits/5776c43830cb687b3c43a62b9294b220e32ae86f to handle this

I needed the first commit so I can fallback to `self.logger` in case something goes wrong while initializing `WithcChannelContext`